clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1197,14 +1181,23 @@ void handleResponse(AbstractResponse abstractResponse) {
                         continue;
                     }
                     boolean isInternal = cluster.internalTopics().contains(topicName);
-                    TreeMap<Integer, TopicPartitionInfo> partitions = new TreeMap<>();
                     List<PartitionInfo> partitionInfos = cluster.partitionsForTopic(topicName);
+                    List<TopicPartitionInfo> partitions = new ArrayList<>(partitionInfos.size());
                     for (PartitionInfo partitionInfo : partitionInfos) {
+                        Node leader = null;
+                        if (partitionInfo.leader() != null || partitionInfo.leader().id() != Node.noNode().id())
```

Comment:
Do you mean a static function in this class or somewhere else? Yes, I did this and the controller node change shortly before I had to step out and clearly didn't check them carefully, as both had bugs. :)